cfme/tests/cloud/test_quota.py

```diff
@@ -0,0 +1,326 @@
+# -*- coding: utf-8 -*-
+import fauxfactory
+import pytest
+from riggerlib import recursive_update
+
+from cfme import test_requirements
+from cfme.base.credential import Credential
+from cfme.cloud.provider.azure import AzureProvider
+from cfme.cloud.provider.openstack import OpenStackProvider
+from cfme.utils.generators import random_vm_name
+
+from widgetastic.utils import partial_match
+
+
+pytestmark = [
+    test_requirements.quota,
+    pytest.mark.long_running,
+    pytest.mark.usefixtures("setup_provider"),
+    pytest.mark.provider(
+        [AzureProvider, OpenStackProvider],
+        required_fields=[["provisioning", "image"]],
+        scope="function",
+    ),
+]
+
+
+def new_credential():
+    return Credential(principal='uid{}'.format(fauxfactory.gen_alphanumeric(4)),
+                      secret='redhat')
+
+
+@pytest.fixture
+def vm_name():
+    return random_vm_name(context="quota")
+
+
+@pytest.fixture
+def template_name(provisioning):
+    return provisioning["image"]["name"]
+
+
+@pytest.fixture
+def prov_data(appliance, provider, provisioning, vm_name):
+    if provider.one_of(OpenStackProvider):
```

Comment:
Required:
No need of two checks... make `OpenStack` default and just `update` for `Azure`